dspace-api/src/test/java/org/dspace/content/RelationshipServiceImplTest.java

```diff
@@ -0,0 +1,287 @@
+    }
+
+    @Test
+    public void testCreate() throws Exception {
+        Relationship relationship = relationshipDAO.create(context,new Relationship());
+        context.turnOffAuthorisationSystem();
+        when(authorizeService.isAdmin(context)).thenReturn(true);
+        assertEquals("TestCreate 0", relationship, relationshipService.create(context));
+        MetadataValue metVal = mock(MetadataValue.class);
+        List<MetadataValue> metsList = new ArrayList<>();
+        List<Relationship> leftTypelist = new ArrayList<>();
+        List<Relationship> rightTypelist = new ArrayList<>();
+        Item leftItem = mock(Item.class);
+        Item rightItem = mock(Item.class);
+        RelationshipType testRel = new RelationshipType();
+        EntityType leftEntityType = mock(EntityType.class);
+        EntityType rightEntityType = mock(EntityType.class);
+        testRel.setLeftType(leftEntityType);
+        testRel.setRightType(rightEntityType);
+        testRel.setLeftLabel("Entitylabel");
+        testRel.setRightLabel("Entitylabel");
+        metsList.add(metVal);
+        relationship = getRelationship(leftItem, rightItem, testRel, 0,0);
+        leftTypelist.add(relationship);
+        rightTypelist.add(relationship);
+        when(virtualMetadataPopulator
+                .isUseForPlaceTrueForRelationshipType(relationship.getRelationshipType(), true)).thenReturn(true);
+        when(authorizeService
+                .authorizeActionBoolean(context, relationship.getLeftItem(), Constants.WRITE)).thenReturn(true);
+        when(authorizeService
+                .authorizeActionBoolean(context, relationship.getRightItem(), Constants.WRITE)).thenReturn(true);
+        when(relationshipService.findByItem(context,leftItem)).thenReturn(leftTypelist);
+        when(relationshipService.findByItem(context,rightItem)).thenReturn(rightTypelist);
+        when(leftEntityType.getLabel()).thenReturn("Entitylabel");
+        when(rightEntityType.getLabel()).thenReturn("Entitylabel");
+        when(metVal.getValue()).thenReturn("Entitylabel");
+        when(metsList.get(0).getValue()).thenReturn("Entitylabel");
+        when(relationshipService
+                .findByItemAndRelationshipType(context, leftItem, testRel, true)).thenReturn(leftTypelist);
+        when(itemService.getMetadata(leftItem, "relationship", "type", null, Item.ANY)).thenReturn(metsList);
+        when(itemService.getMetadata(rightItem, "relationship", "type", null, Item.ANY)).thenReturn(metsList);
+        when(relationshipDAO.create(any(), any())).thenReturn(relationship);
+
+        assertEquals("TestCreate 1", relationship, relationshipService.create(context, relationship));
+        assertEquals("TestCreate 2", relationship, relationshipService.create(context, leftItem, rightItem,
+                testRel,0,0));
+
+        context.restoreAuthSystemState();
+    }
+
+    @Test
+    public void testDelete() throws Exception {
+        context.turnOffAuthorisationSystem();
```

Comment:
Again inline comments would be helpful here. There's a lot of content & relationships getting mocked for this test.